Sources/XCTest/Public/Asynchronous/XCTestCase+Asynchronous.swift

```diff
@@ -41,7 +41,7 @@ public extension XCTestCase {
     ///   these environments. To ensure compatibility of tests between
     ///   swift-corelibs-xctest and Apple XCTest, it is not recommended to pass
     ///   explicit values for `file` and `line`.
-    // FIXME: This should have `@MainActor` to match Xcode XCTest, but adding it causes errors in tests authored pre-Swift Concurrency which don't typically have `@MainActor`.
+    @preconcurrency @MainActor
     func waitForExpectations(timeout: TimeInterval, file: StaticString = #file, line: Int = #line, handler: XCWaitCompletionHandler? = nil) {
         precondition(Thread.isMainThread, "\(#function) must be called on the main thread")
         if currentWaiter != nil {
```

Comment:
This function is the only thing which uses the internal `currentWaiter` property on `XCTestCase`, and when I last touched this (in #340), I left a `// FIXME` comment on that property indicating that when we added `@MainActor` here, we could also consider adding `@MainActor` to that property too.

Now that you're doing that in this PR, could we also either add `@MainActor` to `currentWaiter`, or else delete the FIXME comment?